core/iwasm/fast-jit/jit_ir.c

```diff
@@ -382,15 +382,15 @@ jit_cc_init(JitCompContext *cc, unsigned htab_size)
        blocks respectively.  */
     if (!(entry_block = jit_cc_new_basic_block(cc, 0))
         || !(exit_block = jit_cc_new_basic_block(cc, 0)))
-        goto fail;
+        goto fail2;
 
     if (!(cc->exce_basic_blocks =
               jit_calloc(sizeof(JitBasicBlock *) * JIT_EXCE_NUM)))
-        goto fail;
+        goto fail2;
 
     if (!(cc->incoming_insns_for_exec_bbs =
               jit_calloc(sizeof(JitIncomingInsnList) * JIT_EXCE_NUM)))
-        goto fail;
+        goto fail2;
```

Comment:
The code looks very strange, as it goto upper failed label firstly, normally we should goto fail1 firstly but not goto fail2. How about changing to:
```C
    if (!(entry_block = jit_cc_new_basic_block(cc, 0)))
        goto fail;

    if (!(exit_block = jit_cc_new_basic_block(cc, 0))) {
        jit_basic_block_delete(entry_block);
        goto fail;
    }

    /* Record the entry and exit labels, whose indexes must be 0 and 1
       respectively.  */
    cc->entry_label = jit_basic_block_label(entry_block);
    cc->exit_label = jit_basic_block_label(exit_block);
    bh_assert(jit_reg_no(cc->entry_label) == 0
              && jit_reg_no(cc->exit_label) == 1);

    if (!(cc->exce_basic_blocks =
              jit_calloc(sizeof(JitBasicBlock *) * JIT_EXCE_NUM)))
        goto fail;

    if (!(cc->incoming_insns_for_exec_bbs =
              jit_calloc(sizeof(JitIncomingInsnList) * JIT_EXCE_NUM)))
        goto fail;
```

Reply:
Ok, I have changed it.